### PR TITLE
Add nullability in magellan

### PR DESCRIPTION
**Pull request by narenkmanoharan**
Add better interoperability with Kotlin by adding JSR-305 annotation.

- @ParametersAreNonnullByDefault
- @Nullable
- @NonNull

References: 

- https://medium.com/square-corner-blog/rolling-out-nullable-42dd823fbd89
- https://medium.com/square-corner-blog/non-null-is-the-default-58ffc0bb9111

### CODE REVIEW

File: magellan-library/src/main/java/com/wealthfront/magellan/Screen.java [hunk 1]
```diff
@@ -143,7 +146,7 @@ protected boolean shouldAnimateActionBar() {
     return true;
   }
 
-  public String getTitle(Context context) {
+  public String getTitle(@Nonnull Context context) {
```
Comment (fabiendevos) [inline magellan-library/src/main/java/com/wealthfront/magellan/Screen.java]: why do we need the @Nonnull s since we have @ParametersAreNotNullByDefault?

**Comment by ryanmoelter** [comment]
Do we want to make these nullability annotations a warning to allow people to migrate slowly?

**Comment by fabiendevos** [comment]
@narenkmanoharan can you check what's breaking?

**Comment by jmfayard** [comment]
@fabiendevos I've lot of experience with Magellan and I would say makes this a breaking change and add a small guide on how to upgrade.
The breaking changes are things we really needs to take into account to have a stable app (always check that the activity and the view is not null)

**Pull request closed without merging**